packages/stark-ui/src/modules/date-picker/components/date-picker.component.ts

```diff
@@ -170,8 +381,68 @@ export class StarkDatePickerComponent extends AbstractStarkUiComponent implement
 	 * @param event - The MatDatepickerInputEvent to re-emit
 	 */
 	public onDateChange(event: MatDatepickerInputEvent<moment.Moment>): void {
-		if (event.value) {
-			this.dateChanged.emit(event.value.toDate());
+		this._onTouched();
+
+		const value: Date | undefined = event.value ? event.value.toDate() : undefined;
+		this.dateChange.emit(value);
+		this._onChange(value);
+		this._onValidatorChange();
+	}
+
+	/**
+	 * Handle the date changed on mat-datepicker and emit the dateChanged event
+	 * @param event - The MatDatepickerInputEvent to re-emit
+	 */
+	public onDateInput(event: MatDatepickerInputEvent<moment.Moment>): void {
+		const value: Date | undefined = event.value ? event.value.toDate() : undefined;
+		this.dateInput.emit(value);
+	}
+
+	/**
+	 * Get the `starkTimestampMask` configuration.
+	 * If the inputMask is not enabled, it returns `undefined` to disable `starkTimestampMask`.
+	 * Otherwise, it returns the defined configuration.
+	 */
+	public getTimestampMaskConfig(): StarkDatePickerMaskConfig {
+		if (!this.inputMaskEnabled) {
```

Comment:
Isn't this the same as `if(this.inputMaskEnabled){ return this.maskConfig; }` ? And the type should be `StarkDatePickerMaskConfig | void` I believe.